router/utils/utils.go

```diff
@@ -92,3 +95,60 @@ func EnhanceJsonWithTime(t time.Time, key string, resp []byte) []byte {
 func IsNotEmptyString(s string) bool {
 	return len(strings.TrimSpace(s)) > 0
 }
+
+type Drainer interface {
+	Drain(
+		job *jobsdb.JobT,
+		jobParams JobParameters,
+		destinationsMap map[string]*DestinationWithSources,
+	) (bool, string)
+}
+
+func NewDrainer(conf *config.Config) Drainer {
+	return &drainer{
+		destinationIDs: conf.GetReloadableStringSliceVar(
+			[]string{},
+			"Router.toAbortDestinationIDs",
+		),
+		jobRunIDs: conf.GetReloadableStringSliceVar(
+			[]string{},
+			"RSources.toAbortJobRunIDs",
+		),
+	}
+}
+
+type drainer struct {
+	destinationIDs misc.ValueLoader[[]string]
+	jobRunIDs      misc.ValueLoader[[]string]
+}
+
+func (d *drainer) Drain(
+	job *jobsdb.JobT,
+	jobParams JobParameters,
+	destinationsMap map[string]*DestinationWithSources,
+) (bool, string) {
```

Comment:
`Drain` method shall only accept `job *jobsdb.JobT` as a parameter:
- The method's implementation can unmarshal parameters on its own & only the fields that it is interested in
- Whether a destination is enabled or not, this can be provided through an external function while creating a new drainer, e,.g. `enabledDestinationFunc func(destinationID string) (enabled bool)`

Reply:
why even pass `enabledDestinationFunc`?
if the router worker already knows destination is disabled, it can abort a job right away.
Only otherwise can it check `rt.drainer.Drain(...)`

Reply:
agree about the only-job-as-parameter part